code/datums/datacore.dm

```diff
@@ -95,6 +96,28 @@ GLOBAL_DATUM_INIT(data_core, /datum/datacore, new)
 			crimes |= crime
 			return
 
+/**
+ * Marks a crime as served for on the security record.
+ *
+ * Is used change the status of a single crime between "Served" and "Not Served" someone's security record.
+ * Arguments:
+ * * id - record id.
+ * * cDataId - id of already existing crime.
+ */
+/datum/datacore/proc/serveCrime(id, cDataId)
+	for(var/datum/data/record/checked_record in security)
+		if(checked_record.fields["id"] != id)
+			continue
+		var/list/crimes = checked_record.fields["crim"]
+		for(var/datum/data/crime/crime as anything in crimes)
+			if(crime.dataId != text2num(cDataId))
+				continue
+			if(crime.crimeStatus == "Not Served")
+				crime.crimeStatus = "Served"
+			else
+				crime.crimeStatus = "Not Served"
+			return
```

Comment:
this return will mean only the first crime in the list will be shown, delete the line

Reply:
Once we get the ID match and change the status we don't bother running the other crimes because we already know they're not the one we clicked on. The block with the return only runs if the ID matches the one we want and the `if` check for the continue is false. Am I missing something here?

Reply:
that's what the continue is for, the return will just make it harder to tell if something is broken in the future.

Reply:
Well without the continue we'd still check the IDs of any crimes later on the list against the desired one. I suppose it doesn't make too much of a different for optimization, so sure. Is `break`ing the look any better or does it obscure problems in the same way?

Reply:
Shouldn't we already only have one crime in particular via ``cDataId``?